src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -28,92 +31,143 @@ public ToolManifestFinder(DirectoryPath probStart, IFileSystem fileSystem = null
+                        {
+                            result.Add(p);
+                        }
                     }
 
-                    if (deserializedManifest.version != 1)
+                    if (deserializedManifest.isRoot)
                     {
-                        errors.Add(string.Format("Tools manifest format version {0} is not supported.",
-                            deserializedManifest.version));
+                        return result;
                     }
+                }
+            }
 
-                    foreach (var tools in deserializedManifest.tools)
-                    {
-                        var packageLevelErrors = new List<string>();
-                        var packageIdString = tools.Key;
-                        var packageId = new PackageId(packageIdString);
+            if (!findAnyManifest)
+            {
+                throw new ToolManifestCannotFindException(
+                    string.Format(LocalizableStrings.CannotFindAnyManifestsFileSearched,
+                        string.Join(Environment.NewLine, allPossibleManifests.Select(f => f.Value))));
+            }
 
-                        string versionString = tools.Value.version;
-                        NuGetVersion version = null;
-                        if (versionString is null)
-                        {
-                            packageLevelErrors.Add(LocalizableStrings.MissingVersion);
-                        }
-                        else
-                        {
-                            var versionParseResult = NuGetVersion.TryParse(
-                                versionString, out version);
+            return result;
+        }
 
-                            if (!versionParseResult)
-                            {
-                                packageLevelErrors.Add(string.Format(LocalizableStrings.VersionIsInvalid, versionString));
-                            }
-                        }
+        private SerializableLocalToolsManifest DeserializableLocalToolsManifest(FilePath possibleManifest)
+        {
+            return JsonConvert.DeserializeObject<SerializableLocalToolsManifest>(
+                _fileSystem.File.ReadAllText(possibleManifest.Value), new JsonSerializerSettings
+                {
+                    MissingMemberHandling = MissingMemberHandling.Ignore
+                });
+        }
 
-                        if (tools.Value.commands == null
-                            || (tools.Value.commands != null && tools.Value.commands.Length == 0))
-                        {
-                            packageLevelErrors.Add(LocalizableStrings.FieldCommandsIsMissing);
-                        }
+        private List<ToolManifestPackage> GetToolManifestPackageFromOneManifestFile(
+            SerializableLocalToolsManifest deserializedManifest, FilePath path)
+        {
+            List<ToolManifestPackage> result = new List<ToolManifestPackage>();
```

Comment:
Why did you use the full type in one declaration and var in the other in this case?

Reply:
fixed, changed to one statement

Reply:
I just follow our current editorconfig, it will not warn when var is used in obvious way.

Reply:
That's fine. I was curious though why you are adding two new Lists here and one of them you use var and the other you use the explicit type. It was not saying don't use var or questioning our rules, but more why differ in declaration in this particular case.